apps/nova/src/components/dashboard/PastLessons.tsx

```diff
@@ -11,27 +11,57 @@ import { InView } from "react-intersection-observer";
 import dayjs from "dayjs";
 import { Loading } from "@litespace/luna/Loading";
 
-function asLessons(list: ILesson.FindUserLessonsApiResponse["list"] | null) {
+function asLessons(
+  list: ILesson.FindUserLessonsApiResponse["list"] | null,
+  userRole?: IUser.Role
+) {
   if (!list) return [];
 
   return list.map((lesson) => {
     const teacher = lesson.members.find(
-      (member) => member.role === IUser.Role.Tutor
+      (member) =>
+        member.role === IUser.Role.Tutor ||
+        member.role === IUser.Role.TutorManager
     );
+
+    const student = lesson.members.find(
+      (member) => member.role === IUser.Role.Student
+    );
```

Comment:
You don't need to find both the tutor and the student. You can just find the other member. 


```suggestion
  return list.map((lesson) => {
    const otherMember = lesson.members.find(
      (member) => member.role === userRole 
    );
```